packages/backend-dynamic-feature-service/report.api.md

```diff
@@ -68,6 +68,32 @@ export interface BaseDynamicPlugin {
   version: string;
 }
 
+// @public (undocumented)
+export class CommonJSModuleLoader implements ModuleLoader {
```

Comment:
I agree that, having been fixed for the default use-case, the `CommonJSModuleLoader` could still be considered the default implementation of the `ModuleLoader`, not exported, and just used when no module loader is explicitly provided.

However, this would force rewriting a complete `CommonJSModuleLoader` (most probably duplicating code) as soon as dynamic plugin package resolution rules would be slightly different (what happened for us on RHDH).

This is why, through  the `CommonJSModuleLoaderOptions` (`dynamicPluginPackageNameSuffixes`, and `customResolveDynamicPackage`), as mentioned in the PR description, 
> it also allows customizing this search, in order to support cases when the name of the dynamic plugin (after some repackaging for example) doesn't match exactly the name expected by the code that called `resolvePackagePath`.

By design the dynamic plugins manager **_do not enforce or expect any specific pre-requisite about how the dynamic plugin should be packaged_** (apart from being a Javascript package with a `package.json`). Thus it is not excluded that, in some use-case, dynamic plugin packages would be repackaged with a derived package name (typically a suffixed name), or even embedded as a dependency inside a wrapper package for some specific needs.
Indeed the feedback of RHDH production usage proved that both are useful according to the use-case.

So here's the purpose of exporting `CommonJSModuleLoader`:
- still have it as a default value when no module loader is provided
- but also provide a basis implementation that can be configured to implement custom package resolution rules, according to the dynamic plugin packaging approaches that should be supported.